src/_pytest/recwarn.py

```diff
@@ -167,7 +167,7 @@ def warns(  # noqa: F811
             return func(*args[1:], **kwargs)
 
 
-class WarningsRecorder(warnings.catch_warnings):
+class WarningsRecorder(warnings.catch_warnings):  # type:ignore[type-arg]
```

Comment:
The correct type is `warnings.catch_warnings[list[WarningMessage]]` however that will fail at runtime (because `catch_warnings` doesn't take a `[]`) so LGTM.

Reply:
Yep, that was my reasoning too (we could make it work at runtime using `if TYPE_CHECKING` but not worth the hassle I think). Thanks!